src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,94 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
-
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
-
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+import { useState } from 'react';
+import { getTodos } from '../../api';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  setQuery: (query: string) => void,
+  query: string,
+  setTodos: (todos: Todo[]) => void,
+};
+
+export const TodoFilter: React.FC<Props> = ({
+  setQuery,
+  query,
+  setTodos,
+}) => {
+  const [filter, setFilter] = useState<boolean | undefined>(undefined);
+  const handleFilterChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const selectedValue = event.target.value;
+
+    if (selectedValue === 'active') {
+      getTodos(false).then(setTodos);
+      setFilter(false);
+    } else if (selectedValue === 'completed') {
+      getTodos(true).then(setTodos);
+      setFilter(true);
+    } else {
+      getTodos().then(setTodos);
+      setFilter(undefined);
+    }
+  };
```

Comment:
I think it's worth trying to solve it in a slightly different way. In the App component, you called the getTodos function using useEffect only once during the initial rendering. This means you can filter the received array here instead of calling getTodos every time.

Reply:
Good point, it's not a good idea to make API calls if we don't really need this. Try to refactor your solution, to make API request only 1 time and then just filter response when you need it